### PR TITLE
[Merged by Bors] - chore(Tactic/{Positivity,NormNum}): remove support for non-canonical spellings

### PR DESCRIPTION
**Pull request by eric-wieser**
`Mul.mul` or `Nat.mul` etc should not ever appear in a goal, so there is little reason for these tactics to support them.
If these are in your goal, then something has already gone wrong and `norm_num` isn't responsible for saving you; that's `simp` or `dsimp`'s job.

[Zulip thread](https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/.2310344.20removing.20norm_num.20support.20for.20Nat.2Esub.2C.20Sub.2Esub.2C.20etc/near/420507739)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: test/norm_num.lean [hunk 1]
```diff
@@ -521,18 +521,6 @@ section
...
-example : Add.add 10 2 = 12 := by norm_num1
-example : Nat.sub 10 1 = 9 := by norm_num1
-example : Nat.mod 10 5 = 0 := by norm_num1
-example : Sub.sub 10 1 = 9 := by norm_num1
-example : Sub.sub 10 (-2) = 12 := by norm_num1
-example : Mul.mul 10 1 = 10 := by norm_num1
-example : (Div.div 10 1 : ℚ) = 10 := by norm_num1
-
-end Transparency
-
```
Comment (eric-wieser) [inline test/norm_num.lean:535]: These were added without explanation in https://github.com/leanprover-community/mathlib4/commit/60790d4fd15e2d67606041b9f7058872911a8ac8#r138404530

Comment (ericrbg) [reply test/norm_num.lean:535]: I have a feeling they came up during testing and they thought it'd just be easy to support them.

**Comment by ericrbg** [comment]
I wonder if some of what I'm doing in #10293 counts for this.

File: test/norm_num.lean (hunk 1 above)
Comment (eric-wieser) [reply test/norm_num.lean:535]: It's possible that refactors elsewhere made this behavior no longer necessary

**Review by Vierkantor — APPROVED** [review]
The fact that everything still builds after this change shows we don't need the looser matching functionality. I only added the `Mod.mod` and `Int.emod` matching to the `evalIntMod` functionality because I cargo-culted this together. So I am a bit hesitant of breaking usage downstream, but in summary I think this is the right way to go.

bors r+

**Comment by eric-wieser** [comment]
> The fact that everything still builds after this change shows we don't need the looser matching functionality. I only added the `Mod.mod` and `Int.emod` matching to the `evalIntMod` functionality because I cargo-culted this together. So I am a bit hesitant of breaking usage downstream, but in summary I think this is the right way to go.
> 
> bors r+

Note that I left `Int.emod` and similar behind, as there is possibility for contention there over which operator is "the" modulus operator; downstream code might genuinely use those prefixed definitions to avoid ambiguity.

**Comment by Vierkantor** [comment]
Indeed, I recall the dispute between mathlib and core lean about dividing by negative numbers :)

**Pull request closed without merging**